internal/internal_task_pollers.go

```diff
@@ -242,27 +242,22 @@ func (wtp *workflowTaskPoller) ProcessTask(task interface{}) error {
 	return nil
 }
 
-type decisionTaskPollRequest struct {
-	request      *s.PollForDecisionTaskRequest
-	isStickyTask bool
-}
-
-func (wtp *workflowTaskPoller) release(isStickyTask bool) {
+func (wtp *workflowTaskPoller) release(kind s.TaskListKind) {
 	if wtp.disableStickyExecution {
 		return
 	}
 
 	wtp.requestLock.Lock()
-	if isStickyTask {
+	if kind.Equals(s.TaskListKindWorker) {
```

Comment:
use == to compare for enum type

Reply:
changed, what's the benefits of using ==? Equals function itself use == inside.